monai/apps/auto3dseg/ensemble_builder.py

```diff
@@ -0,0 +1,227 @@
+        predict results after the models are ranked/weighted
+        """
+        raise NotImplementedError
+
+
+class EnsembleBestN(ScriptEnsemble):
+    """
+    Ensemble method that select N model out of all using the models' best_metric scores
+
+    Args:
+        n_best: number of models to pick for ensemble (N).
+    """
+
+    def __init__(self, n_best: int = 5):
+
+        super().__init__()
+        self.n_best = n_best
+
+    def sort_score(self):
+        """sort the best_metrics"""
+        scores = concat_val_to_np(self.algos, [ScriptEnsembleKeys.SCORE])
+        return np.argsort(scores).tolist()
+
+    def rank_algos(self):
+        """rank the model"""
+        ranks = self.sort_score()
+        if len(ranks) < self.n_best:
+            raise ValueError("Number of available algos is less than user-defined N")
+
+        # get the indices that the rank is larger than N
+        indices = [i for (i, r) in enumerate(ranks) if r >= self.n_best]
+
+        # remove the found indices
+        indices = sorted(indices, reverse=True)
+
+        self.ranked_algo = deepcopy(self.algos)
+        for idx in indices:
+            if idx < len(self.ranked_algo):
+                self.ranked_algo.pop(idx)
+
+    def predict(self, param: Optional[Dict[str, Any]] = None):
+        """predict the ensembled pred"""
+        if param is None:
+            param = {}
+
+        for i in range(len(self.infer_files)):
+            preds = []
+            infer_filename = self.infer_files[i]
+            for algo in self.ranked_algo:
+                infer_instance = algo[ScriptEnsembleKeys.ALGO]
+                param.update({"files": [infer_filename]})
+                pred = infer_instance.predict(param)
+                preds.append(pred[0])
+            output = sum(preds) / len(preds)
+            print("ensemble_algorithm, preds", len(preds), "output shape", output.shape)
+
+
+class ScriptEnsembleBuilder:
+    """
+    Build ensemble workflow from configs and arguments.
+
+    Args:
+        history: a collection of trained bundleAlgo algorithms.
+        data_src_cfg_filename: filename of the data source.
+
+    Examples:
+
+        ..code-block:: python
+            builder = ScriptEnsembleBuilder(history, data_src_cfg)
+            builder.set_ensemble_method(EnsembleBestN(3))
+            ensemble = builder.get_ensemble()
+
+            result = ensemble.predict()
+    """
+
+    def __init__(self, history: Sequence[Dict], data_src_cfg_filename: Optional[str] = None):
+        self.infer_algos: List[Dict[ScriptEnsembleKeys, Any]] = []
+        self.ensemble: ScriptEnsemble
+        self.data_src_cfg = ConfigParser(globals=False)
+
+        if data_src_cfg_filename is not None and os.path.exists(str(data_src_cfg_filename)):
+            self.data_src_cfg.read_config(data_src_cfg_filename)
+
+        for h in history:
+            # load inference_config_paths
+            # raise warning/error if not found
+            if len(h.keys()) > 1:
```

Comment:
could be simplified:
```suggestion
            if len(h) > 1:
```

Reply:
Fixed. Thanks!